### DISCUSSION TITLE
Don't register URI scheme when Discord RPC is turned off in settings

### DISCUSSION THREAD

**Discussion by sineplusx** [ppy/osu#27856] [Ideas] [answered]
Support for Discord game invites in multiplayer lobbies was added in #27443.
For the invites to work properly, a URI scheme is automatically created on every game launch. This happens regardless of whether the user has set the Discord Rich Presence feature in the settings to "Off".

I would like the URI scheme to not be created in the case the setting is "Off", as it pollutes the associations with a useless entry if you don't have a local Discord installation.

**Comment by NiceAesth** [comment]
just adding my two cents but as stable has shown, other developers have made use of osu! URIs. removing multiplayer invite URI handling behind an opaque discord presence setting would be pretty stupid to keep track of for the end user (not to mention that having that association literally has no impact on you at all)

**Reply by sineplusx** [reply]
I think you misunderstood. The URI I refer to (`discord-1216669957799018608://`) is only used for the discord invite functionality introduced in the PR. The code is here:

https://github.com/ppy/osu/blob/1c02cbfa5dafc2138d64220899832fb2ef5a465d/osu.Desktop/DiscordRichPresence.cs#L87

I dont think this is used or will be used by other developers. If this didn't have "an impact" on me I wouldnt have made a discussion. The reason why I noticed at all is that since the new release I get an error on each launch, since it tries adding the association but is blocked by my security profiling.

MIME type associations on Linux are stored with the corresponding apps as `.desktop` files, and the AppImage ships with one already. Adding another association in a seperate file at runtime feels jank, I just like to keep my files clean.

**Reply by peppy** [reply]
Seems like it should be special cased out for linux, which the URI being added to the appimage spec instead?

**Comment by sineplusx** [comment, marked as answer]
Closing in favor of tracking in https://github.com/ppy/osu/issues/34631

**Reply by n0099** [reply]
That issue is about creating desktop entry file in `~/.local/share/applications`, not register url scheme in `~/.config/mimeapps`
